app/controllers/assignments_controller.rb

```diff
@@ -344,17 +344,24 @@ def batch_runs
   # Return assignment grade distributions
   def grade_distribution
     assignment = record
+    assignment_remark_requests = assignment.groupings.joins(current_submission_used: :submitted_remark)
     summary = {
       name: "#{assignment.short_identifier}: #{assignment.description}",
-      average: assignment.results_average || 0,
-      median: assignment.results_median || 0,
+      average: assignment.results_average(point_mark: true) || 0,
+      median: assignment.results_median(point_mark: true) || 0,
+      max_mark: assignment.max_mark || 0,
+      standard_deviation: assignment.results_standard_deviation || 0,
       num_submissions_collected: assignment.current_submissions_used.size,
       num_submissions_graded: assignment.current_submissions_used.size -
         assignment.ungraded_submission_results.size,
       num_fails: assignment.results_fails,
       num_zeros: assignment.results_zeros,
       groupings_size: assignment.groupings.size,
-      num_outstanding_remark_requests: assignment.outstanding_remark_request_count
+      num_students_with_submitted_work: assignment.groupings.joins(:accepted_students).count,
+      num_active_students: assignment.course.students.active.size,
+      num_remark_requests: assignment_remark_requests.count,
+      num_remark_requests_completed: assignment_remark_requests.count -
```

Comment:
I could also get this attribute by doing something like `assignment_remark_requests.where('results.marking_state': :complete).count`. This would eliminate the need for the `outstanding_remark_request_count` attribute and it can be safely removed from the Assignment model.

It seemed a little drastic to do though without a second opinion. Do you think that sounds like a good idea or is this current approach better?